src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.js

```diff
@@ -289,7 +293,7 @@ export const CloudEnvironmentModal = ({
     const cookieReady = Utils.cond(
       [cloudProvider === cloudProviderTypes.AZURE && toolLabel === appToolLabels.CROMWELL, () => azureCookieReady.readyForCromwellApp],
       [Utils.DEFAULT, () => leoCookieReady])
-    const isDisabled = !doesCloudEnvForToolExist || !cookieReady || !canCompute || busy || isToolBusy || !isLaunchSupported(toolLabel)
+    const isDisabled = !doesCloudEnvForToolExist || !cookieReady || !canCompute || busy || isToolBusy || !isLaunchSupported(toolLabel) || !doesWorkspaceSupportCromwellApp(workspace?.workspace?.createdDate, cloudProvider, toolLabel)
```

Comment:
Hmm... should we only disable the launch button if `!doesWorkspaceSupportCromwellApp && [it is a cromwell app]`?

Reply:
Ah I see... method name a bit misleading since if its anything other than cromwell it returns true.  Could we perhaps make `isLaunchSupported` a bit more robust, moving it to `tool-utils.ts` and giving it the same treatment as `isToolHidden` where we pass `cloudProvider` (and in this case possibly the workspace with created date) to the method?

Reply:
Yes. `doesWorkspaceSupportCromwellApp` does ensure that we do this workspace date creation check only for Cromwell app (see [here](https://github.com/DataBiosphere/terra-ui/blob/95fd75ebb722732060a2f1fc71038af355f49ddf/src/pages/workspaces/workspace/analysis/utils/app-utils.ts#L32)). For all other apps in Azure or for anything in GCP it returns true

Reply:
I think that's possible but I am not sure how to add [this condition](https://github.com/DataBiosphere/terra-ui/blob/95fd75ebb722732060a2f1fc71038af355f49ddf/src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.js#L308) if logic for `doesWorkspaceSupportCromwellApp` was merged within `isLaunchSupported`. I would need a different boolean value for this condition so that this very specific message only in this case is visible to the user 🤔

Reply:
Another reason I had this in a separate function is that when `doesWorkspaceSupportCromwellApp` returns false, in Context Bar we don't want to display Cromwell app icon. So this logic would not be solely for whether the app is launch-able or not (if combined with `isLaunchSupported` function).